crates/wash-lib/src/cli/inspect.rs

```diff
@@ -85,32 +92,51 @@ pub async fn handle_command(
         .await?;
     }
 
-    if is_wasm(&buf) {
-        // Inspect an actor module
-        let module_name = command.target.clone();
-        let jwt_only = command.jwt_only;
-        let caps = get_caps(command.clone(), &buf).await?;
-        let token =
-            caps.ok_or_else(|| anyhow!("No capabilities discovered in : {}", module_name))?;
-        if jwt_only {
-            let out = CommandOutput::from_key_and_text("token", token.jwt);
-            Ok(out)
-        } else {
-            let validation = wascap::jwt::validate_token::<Actor>(&token.jwt)?;
-            let out = render_actor_claims(token.claims, validation);
-            Ok(out)
+    let output = match wasmparser::Parser::new(0).parse_all(&buf).next() {
+        // Inspect the WIT of a Wasm component
+        Some(Ok(wasmparser::Payload::Version {
+            encoding: wasmparser::Encoding::Component,
+            ..
+        })) if command.wit => {
+            let witty = wit_component::decode(&buf).expect("Failed to decode WIT");
+            let resolve = witty.resolve();
+            let main = witty.package();
+            let mut printer = wit_component::WitPrinter::default();
+            CommandOutput::from_key_and_text(
+                "wit",
+                printer
+                    .print(resolve, main)
+                    .expect("should be able to print WIT world from a component"),
+            )
         }
-    } else {
-        handle_provider_archive(command.clone(), &buf).await
-    }
-}
-
-/// Checks if the given file is a wasm file by searching a WASM magic number at the start of a binary
-fn is_wasm(input: &[u8]) -> bool {
-    if input.len() < 4 {
-        return false;
-    }
-    input[0..4] == WASM_MAGIC
+        // Catch trying to inspect a WIT from a WASI Preview 1 module
+        Some(Ok(wasmparser::Payload::Version {
+            encoding: wasmparser::Encoding::Module,
+            ..
+        })) if command.wit => {
+            bail!("No WIT present in Wasm, this looks like a WASI Preview 1 module")
+        }
+        // Inspect claims inside of Wasm
+        Some(Ok(_)) => {
```

Comment:
When does this happen?

Reply:
This happens whenever people don't supply the `--wit` flag, e.g. intending to inspect the claims of the Wasm

Reply:
Basically this was properly parsed as a WebAssembly module or component, but `--wit` wasn't supplied, so we're going through the previous behavior